### PR TITLE
HW-52629, HW-52630

### PR DESCRIPTION
**Pull request by oleksii-lubianyi-epam**

### CODE REVIEW

**Comment by oleksii-lubianyi-epam** [comment]
@syyang-hw I've provided fixes you've mentioned in previous PR

**Comment by oleksii-lubianyi-epam** [comment]
NB: Should be merged together with https://github.com/hyperwallet/hyperwallet-ios-ui-sdk/pull/26

File: Sources/Model/HyperwalletBankAccount.swift [hunk 1]
```diff
@@ -64,30 +111,39 @@ public final class HyperwalletBankAccount: HyperwalletTransferMethod {
         /// - Parameters:
         ///   - transferMethodCountry: The bank account country.
         ///   - transferMethodCurrency: The bank account currency.
-        ///   - transferMethodProfileType: The method profile type
-        public init(transferMethodCountry: String, transferMethodCurrency: String, transferMethodProfileType: String) {
+        ///   - transferMethodProfileType: The bank account holder's profile type 'ProfileType'
+        public init(transferMethodCountry: String, transferMethodCurrency: String, transferMethodProfileType: ProfileType) {
```
Comment (jsaini-hw) [inline Sources/Model/HyperwalletBankAccount.swift]: keep it as String instead of ProfileType

Comment (oleksii-lubianyi-epam) [reply Sources/Model/HyperwalletBankAccount.swift]: Reverted

**Comment by oleksii-lubianyi-epam** [comment]
@jsaini-hw 
> Add the newly added fields in the HyperwalletBankAccountTests. Split existing test testCreateBankAccount_success() to testCreateBankAccount_individual_success() and testCreateBankAccount_business_success(). Add individual fields in the testCreateBankAccount_individual_success() and business related fields in testCreateBankAccount_business_success()

Done

File: Tests/BankAccount/Business/HyperwalletBankAccountBusinessTests.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        // When
+        let bankAccount = HyperwalletBankAccount
+            .Builder(transferMethodCountry: "US",
+                     transferMethodCurrency: "USD",
+                     transferMethodProfileType: "BUSINESS")
+            .bankAccountId("7861012345")
+            .branchId("102000021")
+            .bankAccountRelationship(.ownCompany)
+            .bankAccountPurpose(.checking)
+            .build()
```
Comment (jsaini-hw) [inline Tests/BankAccount/Business/HyperwalletBankAccountBusinessTests.swift]: set profile fields as well for Business and Individual Tests

File: Tests/BankAccount/Individual/HyperwalletBankAccountIndividualTests.swift [hunk 3]
```diff
@@ -14,10 +14,10 @@ class HyperwalletBankAccountTests: XCTestCase {
         }
     }
 
-    func testCreateBankAccount_success() {
+    func testCreateBankAccount_individual_success() {
```
Comment (jsaini-hw) [inline Tests/BankAccount/Individual/HyperwalletBankAccountIndividualTests.swift]: Add profile fields such as firstName, lastName etc etc in the test

File: Tests/Responses/BankAccount/BankAccountBusinessResponse.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    "city": "Test City",
+    "stateProvince": "WA",
+    "country": "US",
+    "postalCode": "12345",
+    "links": [
+        {
+            "params": {
+                "rel": "self"
+            },
+            "href": "https://qamaster-hyperwallet.aws.paylution.net/rest/v3/users/usr-1297a088-230c-4487-a5b6-48eb9a1cd1b9/bank-accounts/trm-61231a80-a31f-4c44-b4b5-ece158036647"
```
Comment (jsaini-hw) [inline Tests/Responses/BankAccount/BankAccountBusinessResponse.json]: href should never contain actual url and data. Update it to 
```
https://localhost/rest/v3/users/usr-1234/bank-accounts/trm-1234
```

Comment (oleksii-lubianyi-epam) [reply Tests/Responses/BankAccount/BankAccountBusinessResponse.json]: Reviewed all the mocked responses and replaced all the real data (urls and tokens) with dummy one.

File: Tests/BankAccount/Business/HyperwalletBankAccountBusinessTests.swift [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            errorResponse = error
+            expectation.fulfill()
+        })
+        wait(for: [expectation], timeout: 1)
+
+        // Then
+        XCTAssertNotNil(bankAccount)
+        XCTAssertNil(errorResponse, "The `errorResponse` should be nil")
+        XCTAssertNotNil(bankAccountResponse?.getFields())
+        XCTAssertEqual(bankAccountResponse?.getField(fieldName: .city) as! String, "Test City")
```
Comment (oleksii-lubianyi-epam) [inline Tests/BankAccount/Business/HyperwalletBankAccountBusinessTests.swift]: @jsaini-hw @syyang-hw 
Should all the fields set above be validated in response here?

File: Tests/BankAccount/Business/HyperwalletBankAccountBusinessTests.swift (hunk 2 above)
Comment (oleksii-lubianyi-epam) [reply Tests/BankAccount/Business/HyperwalletBankAccountBusinessTests.swift]: Done

File: Tests/BankAccount/Individual/HyperwalletBankAccountIndividualTests.swift (hunk 3 above)
Comment (oleksii-lubianyi-epam) [reply Tests/BankAccount/Individual/HyperwalletBankAccountIndividualTests.swift]: Added missing fields, thanks

File: Tests/BankAccount/Business/HyperwalletBankAccountBusinessTests.swift (hunk 5 above)
Comment (jsaini-hw) [reply Tests/BankAccount/Business/HyperwalletBankAccountBusinessTests.swift]: @oleksii-lubianyi-epam yes

**Pull request merged**